src/http/server.rs

```diff
@@ -559,9 +578,7 @@ impl<'a> EspHttpConnection<'a> {
     pub fn uri(&self) -> &str {
         self.assert_request();
 
-        let c_uri = unsafe {
-            CStr::from_bytes_with_nul_unchecked(mem::transmute(self.request.0.uri.as_slice()))
-        };
+        let c_uri = unsafe { CStr::from_ptr(self.request.0.uri.as_ptr()) };
```

Comment:
This was a bug, it violated the `CStr` safety contract, and ended up as a 512 byte long mostly-zeroed `&str`

Reply:
OK - thanks for fixing!